### PR TITLE
Fix Round()

### PR DESCRIPTION
**Pull request by StefH**

### CODE REVIEW

**Comment by Lej** [comment]
This seems to fix round for 2.5 case, but round still fails for negative numbers when `MidpointRounding.ToEven` causes rounding towards negative infinity.

Below expanded unit test where the case [InlineData("-3.5", -3)] fails (it is rounded to -4 when it should be rounded to -3). Note that I changed `string xpath = "round(number(value))";` to `string xpath = "round(xs:double(value))";` to test some edge cases.

```
        // https://www.w3.org/TR/xpath-functions/#func-round
        [Theory]
        [InlineData("2.51", 3)]
        [InlineData("2.5", 3)]
        [InlineData("2.4999", 2)]
        [InlineData("0", 0)]
        [InlineData("-0", 0)]
        [InlineData("-2.4999", -2)]
        [InlineData("-2.5", -2)]
        [InlineData("-2.51", -3)]
        [InlineData("-3.5", -3)]
        [InlineData("INF", double.PositiveInfinity)]
        [InlineData("-INF", double.NegativeInfinity)]
        [InlineData("NaN", double.NaN)]
        public void XPath2Evaluate_round(string value, object expected)
        {
            // Arrange
            string xpath = "round(xs:double(value))";
            var xml = new XmlDocument { InnerXml = $"<value>{value}</value>" };
            var nav = xml.CreateNavigator();

            // Act
            var result = nav.XPath2Evaluate(xpath);

            // Assert
            result.Should().Be(expected);
        }
```

**Comment by StefH** [comment]
@Lej 
Thanks for your review.

Code has been changed. Please check again.

File: src/XPath2/ExtFuncs.cs [hunk 1]
```diff
@@ -1026,143 +1026,218 @@ public static object AdjustTimeToTimezone(object arg, object tz)
...
+                case double valueAsDouble:
+                    double fractionAsDouble = Math.Abs(valueAsDouble - Math.Floor(valueAsDouble));
+                    return fractionAsDouble >= 0.5 ? Math.Ceiling(valueAsDouble) : Math.Floor(valueAsDouble);
+
+                case decimal valueAsDecimal:
+                    decimal fractionAsDecimal = Math.Abs(valueAsDecimal - Math.Floor(valueAsDecimal));
+                    return fractionAsDecimal >= 0.5m ? Math.Ceiling(valueAsDecimal) : Math.Floor(valueAsDecimal);
+
+                case Integer valueAsInteger:
+                    decimal valueCastToDecimal = (decimal)valueAsInteger;
```
Comment (Lej) [inline src/XPath2/ExtFuncs.cs]: Maybe I'm missing something but if the value is already an integer we don't need to round it right? Could just return valueAsInteger?

**Comment by Lej** [comment]
> @Lej
> Thanks for your review.
> 
> Code has been changed. Please check again.

Looks like it should work now. Had a small comment, not sure if relevant.
